x-pack/test/saved_object_api_integration/common/suites/bulk_get.ts

```diff
@@ -91,16 +95,31 @@ export function bulkGetTestSuiteFactory(esArchiver: any, supertest: SuperTest<an
       const { user, spaceId = SPACES.DEFAULT.spaceId, tests } = definition;
 
       describeFn(description, () => {
-        before(() =>
-          esArchiver.load(
-            'x-pack/test/saved_object_api_integration/common/fixtures/es_archiver/saved_objects/spaces'
-          )
-        );
-        after(() =>
-          esArchiver.unload(
-            'x-pack/test/saved_object_api_integration/common/fixtures/es_archiver/saved_objects/spaces'
-          )
-        );
+        before(async () => {
+          await testDataLoader.before();
+          await testDataLoader.beforeEach([
+            {
+              spaceName: null,
+              dataUrl:
+                'x-pack/test/saved_object_api_integration/common/fixtures/kbn_archiver/default_space.json',
+            },
+            {
+              spaceName: SPACE_1.id,
+              dataUrl:
+                'x-pack/test/saved_object_api_integration/common/fixtures/kbn_archiver/space_1.json',
+            },
+            {
+              spaceName: SPACE_2.id,
+              dataUrl:
+                'x-pack/test/saved_object_api_integration/common/fixtures/kbn_archiver/space_2.json',
+            },
+          ]);
+        });
+
+        after(async () => {
+          await testDataLoader.after();
+          await testDataLoader.afterEach();
+        });
```

Comment:
If my understanding of `after` is correct (that it's basically an `afterAll` function), then I would expect it to run _after_ `afterEach`. So the run order would end up being like this:

1. before
2. beforeEach
3. test
4. afterEach
5. after

However, what's the reason why `afterEach` isn't called in an `afterEach` function? E.g:

```suggestion
        afterEach(testDataLoader.afterEach);
        after(testDataLoader.after);
```

(I think you can just give the functions as arguments without a wrapping function)

The same goes for the `before` function above

Reply:
I've renamed the `TestDataLoader` functions in the latest commit. They are really just to setup data and should be able to be called from any test hook.

Reply:
Ah ok, thanks for explaining 👍